File: Problems/_beginner_submit_tmp/P7012.py
from typing import List


class Solution:
    def minAuditDays(self, loads: List[int]) -> int:
        # 空数组无需抽查
        if not loads:
            return 0
        # 先算总负载；选出之和必须严格大于一半
        total = sum(loads)
        # 贪心：优先抽查负载最大的天，天数才可能最少
        a = sorted(loads, reverse=True)
        s = 0
        for i, x in enumerate(a, 1):
            s += x
            # 2*s > total 等价于 s > total - s
            if s * 2 > total:
                return i
        return len(a)

File: Problems/_beginner_submit_tmp/P7012.java
import java.util.*;

public class Solution {
    public int minAuditDays(int[] loads) {
        // 空数组无需抽查
        if (loads == null || loads.length == 0) {
            return 0;
        }
        // 计算总负载（用 long 防止溢出）
        long total = 0;
        Integer[] a = new Integer[loads.length];
        for (int i = 0; i < loads.length; i++) {
            total += loads[i];
            a[i] = loads[i];
        }
        // 降序排序：优先选大的
        Arrays.sort(a, Collections.reverseOrder());
        long s = 0;
        for (int i = 0; i < a.length; i++) {
            s += a[i];
            // 选出之和严格大于未选之和
            if (s * 2 > total) {
                return i + 1;
            }
        }
        return a.length;
    }
}
